server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -181,6 +176,7 @@ protected final IntFunction<InternalAggregations> buildSubAggsForBuckets(long[]
         for (int i = 0; i < subAggregators.length; i++) {
             aggregations[i] = subAggregators[i].buildAggregations(bucketOrdsToCollect);
         }
+        updateCircuitBreaker("built_sub_aggregations");
```

Comment:
Not sure if we should call this before the aggs building or after. Opinions? I don't think there will be much difference

Reply:
I think we should add it inside the loop, that's the reason we are sampling the calls to the breaker?

Reply:
This loop is just per sub-agg. But after thinking twice, indeed, it's important. Changing!